lib/code_ownership/private/ownership_mappers/file_annotations.rb

```diff
@@ -55,22 +55,28 @@ def file_annotation_based_owner(filename)
           return unless File.file?(filename)
 
           # The annotation should be on line 1 but as of this comment
-          # there's no linter installed to enforce that. We therefore check the
-          # first line (the Ruby VM makes a single `read(1)` call for 8KB),
-          # and if the annotation isn't in the first two lines we assume it
-          # doesn't exist.
-
-          line_1 = File.foreach(filename).first
-
-          return if !line_1
-
-          begin
-            team = line_1[TEAM_PATTERN, :team]
-          rescue ArgumentError => ex
-            if ex.message.include?('invalid byte sequence')
-              team = nil
-            else
-              raise
+          # there's no linter installed to enforce that.
+          # We therefore read each line of the file until the line does not start
+          # with a '#'. We then attempt to parse out the team name from those lines.
+          team = nil
+          comment_lines = []
+
+          File.foreach(filename) do |line|
+            break unless line.start_with?('#')
+            comment_lines << line
+          end
+
+          return unless comment_lines.any?
+
+          comment_lines.each do |line|
+            begin
+              team = line[TEAM_PATTERN, :team]
```

Comment:
If more than one line has the `@team` annotation we'll end up taking the last line where it was defined.

If we wanted to combine teams (for some reason), we'd need to concat the names, etc.
If we wanted the first line only, we'd have to add a break, or similar, after finding our first match.